### ISSUE TITLE
table of new season not consistant with results from previous season

### ISSUE THREAD

**Issue by hoehnp** [Xenoage/Bolzplatz2006#1]
Despite Stuckrat lost the last match, they come in even one rank better in the new season compared to the [table](https://github.com/hoehnp/Unofficial_mirror_bolzplatz2006/raw/master/Bug_after.png) at the last day of the [previous season](https://github.com/hoehnp/Unofficial_mirror_bolzplatz2006/raw/master/Bug_before.png). If I can figure out where the games are saved in windows, I can also provide the  the xml game file.

**Comment by Xenoage** [comment]
Are you sure that the second screenshot shows the ranking after the last matchday? Below I see "17th matchday" on the button; shouldn't it be "18th matchday"?

The XML savegames are stored in `data\savegames\[career|worldcup]`, e.g. `C:\Program Files (x86)\Bolzplatz 2006\data\savegames\...` on my machine.

**Comment by hoehnp** [comment]
The table of the last matchday is not accessible. After the last match, the new season starts immediately. I don't find any user files in Program Files and wonder if someone without administrative rights even could write there.

**Comment by Xenoage** [comment]
Then the problem is probably that the ranking screen is not shown after the last matchday, and we could add that?

Location of savegames:
`File slotFile = new File("data/savegames/career/career" + index + ".xml");`
seems to implicitly use `%UserProfile%\AppData\Local\VirtualStore\Program Files (x86)\Bolzplatz 2006\...` , e.g. `C:\Users\Andreas\AppData\Local\VirtualStore\Program Files (x86)\Bolzplatz 2006\data\savegames\career`

**Comment by hoehnp** [comment]
I believe there are those are two different issues. My claim is that the table of the last match day should be consistent with the one at the start of the season. Having a final table of the season would be a nice additional feature and could maybe help debug this issue.

**Comment by hoehnp** [comment]
I uploaded the respective xml files of last matchday and first match day here: https://gist.github.com/hoehnp/b188f94f1a08d96fb0bc0698f35a78ed

**Comment by Xenoage** [comment]
The "last matchday" file tells me that it's `currentleaguematchday="18"` , which means there is still one game left for this season, isn't it?

I loaded it and matchday 18 has still to be played. We don't show the result of matchday 18, and maybe we also don't save it? Can you further investigate this issue?

**Comment by hoehnp** [comment]
You are right about this. My point was I was winning at the last day against Stuckrad. Yet they managed to improve their ranking.

**Comment by hoehnp** [comment]
Another suggestion for debugging could be to save results from multiple seasons in the xml files. Would that be feasible?

**Comment by Xenoage** [comment]
I think you could have a look into `com.xenoage.bp2k6.gamemodes.Career::matchOver` . There, you could check that the results after a matchday are correctly computed, or you could save the current ranking in temporary files, based on `::saveToXMLFile`.
